src/shared/numa-util.c

```diff
@@ -124,6 +126,58 @@ int numa_to_cpu_set(const NUMAPolicy *policy, CPUSet *ret) {
         return 0;
 }
 
+static int numa_max_node(int *ret) {
```

Comment:
btw, to simplify things, you could use `static int numa_max_node(void)` as prototype for this, and return both errors and the number as return value. i.e. if the function returns < 0 it's an error, and if it returns >= 0 it's the number of nodes